packages/react-reconciler/src/ReactFiberHooks.new.js

```diff
@@ -1545,36 +1533,26 @@ function startTransition(setPending, config, callback) {
   }
 }
 
-function mountTransition(
-  config: SuspenseConfig | void | null,
-): [(() => void) => void, boolean] {
+function mountTransition(): [(() => void) => void, boolean] {
   const [isPending, setPending] = mountState(false);
-  const start = mountCallback(startTransition.bind(null, setPending, config), [
-    setPending,
-    config,
-  ]);
+  // The `start` method can be stored on a ref, since `setPending`
+  // never changes.
+  const start = startTransition.bind(null, setPending);
+  mountRef(start);
   return [start, isPending];
 }
 
-function updateTransition(
-  config: SuspenseConfig | void | null,
-): [(() => void) => void, boolean] {
-  const [isPending, setPending] = updateState(false);
-  const start = updateCallback(startTransition.bind(null, setPending, config), [
-    setPending,
-    config,
-  ]);
+function updateTransition(): [(() => void) => void, boolean] {
+  const [isPending] = updateState(false);
+  const startRef = updateRef();
+  const start: (() => void) => void = (startRef.current: any);
   return [start, isPending];
 }
 
-function rerenderTransition(
-  config: SuspenseConfig | void | null,
-): [(() => void) => void, boolean] {
-  const [isPending, setPending] = rerenderState(false);
-  const start = updateCallback(startTransition.bind(null, setPending, config), [
-    setPending,
-    config,
-  ]);
+function rerenderTransition(): [(() => void) => void, boolean] {
+  const [isPending] = rerenderState(false);
+  const startRef = updateRef();
+  const start: (() => void) => void = (startRef.current: any);
```

Comment:
Any downside to calling this startTransition?

Reply:
There's already a `startTransition` in this module, would have to rename that one to like `startTransitionImpl` or something.